auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -310,16 +312,51 @@ public TypeSpec createTypeAdapter(ClassName className, ClassName autoValueClassN
     return classBuilder.build();
   }
 
+  private static void buildParameterizedToken(MethodSpec.Builder constructor, TypeName typeArg,
+                                              List<TypeVariableName> typeParams, boolean first) {
+    if (!first) {
+      constructor.addCode(", ");
+    }
+    if (typeArg instanceof ParameterizedTypeName) {
+      ParameterizedTypeName paramTypeArg = (ParameterizedTypeName) typeArg;
+      constructor.addCode("$T.newParameterizedTypeWithOwner(null, $T.class", $Gson$Types.class, paramTypeArg.rawType);
```

Comment:
Yeah. Maybe we could add the factory methods to TypeToken though. Moshi
exposes Types for this which is very convenient.

On Fri, Aug 12, 2016 at 11:29 AM Ryan Harter <notifications@github.com>
wrote:

> In
> auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java
> <https://github.com/rharter/auto-value-gson/pull/66#discussion_r74610450>:
>
> > @@ -310,16 +312,51 @@ public TypeSpec createTypeAdapter(ClassName className, ClassName autoValueClassN
> >      return classBuilder.build();
> >    }
> >
> > +  private static void buildParameterizedToken(MethodSpec.Builder constructor, TypeName typeArg,
> > +                                              List<TypeVariableName> typeParams, boolean first) {
> > +    if (!first) {
> > +      constructor.addCode(", ");
> > +    }
> > +    if (typeArg instanceof ParameterizedTypeName) {
> > +      ParameterizedTypeName paramTypeArg = (ParameterizedTypeName) typeArg;
> > +      constructor.addCode("$T.newParameterizedTypeWithOwner(null, $T.class", $Gson$Types.class, paramTypeArg.rawType);
>
> Yeah, my question was basically "is this protected for patch releases",
> but I suppose since it's not public API it's not.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/rharter/auto-value-gson/pull/66/files/036c1f4a9a5caa2bf43e8c31da491c300a3892fe#r74610450>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEEEfIzAJylznkf17-QQgT6fgZXj3MKks5qfJFVgaJpZM4JiIcT>
> .
>